website/source/docs/provisioners/inspec.html.md

```diff
@@ -0,0 +1,135 @@
+---
+description: |
+    The inspec Packer provisioner allows inspec profiles to be run to test the
+    machine.
+layout: docs
+page_title: 'InSpec - Provisioners'
+sidebar_current: 'docs-provisioners-inspec'
+---
+
+# InSpec Provisioner
+
+Type: `inspec`
+
+The `inspec` Packer provisioner runs InSpec profiles. It dynamically creates a
+target configured to use SSH, runs an SSH server, executes `inspec exec`, and
+marshals InSpec tests through the SSH server to the machine being provisioned
+by Packer.
+
+## Basic Example
+
+This is a fully functional template that will test an image on DigitalOcean.
+Replace the mock `api_token` value with your own.
+
+``` json
+{
+  "provisioners": [
+    {
+      "type": "inspec",
+      "profile": "https://github.com/dev-sec/linux-baseline"
+    }
+  ],
+
+  "builders": [
+    {
+      "type": "digitalocean",
+      "api_token": "6a561151587389c7cf8faa2d83e94150a4202da0e2bad34dd2bf236018ffaeeb",
+      "image": "ubuntu-14-04-x64",
+      "region": "sfo1"
+    }
+  ]
+}
+```
+
+## Configuration Reference
+
+Required Parameters:
+
+-   `profile` - The profile to be executed by InSpec.
+
+Optional Parameters:
+
+-   `inspec_env_vars` (array of strings) - Environment variables to set before
+    running InSpec. Usage example:
+
+    ``` json
+    {
+      "inspec_env_vars": [ "AWS_SECRET_KEY_ID=lq9WOb5ME5WxOg5wsIpBz87kh4B8GbZNwd/PDkiq", "AWS_ACCESS_KEY_ID=AKIAI2SN2YBIPZJUW56Q" ]
+    }
+    ```
+
+-   `command` (string) - The command to invoke inspec. Defaults to `inspec`.
+
+-   `extra_arguments` (array of strings) - Extra arguments to pass to InSpec.
+    These arguments *will not* be passed through a shell and arguments should
+    not be quoted. Usage example:
+
+    ``` json
+    {
+      "extra_arguments": [ "--sudo", "--reporter", "json" ]
+    }
+    ```
+
+-   `attributes` (array of strings) - Attribute Files used by InSpec which will
+    be passed to the `--attrs` argument of the `inspec` command when this
+    provisioner runs inspec. Specify this if you want a different location.
+    Note using also `"--attrs"` in `extra_arguments` will override this
+    setting.
+
+-   `attributes_directory` (string) - The directory in which to place the
+    temporary generated InSpec Attributes file. By default, this is the
+    system-specific temporary file location. The fully-qualified name of this
+    temporary file will be passed to the `--attrs` argument of the `inspec`
+    command when this provisioner runs inspec. Specify this if you want a
+    different location.
+
+-   `backend` (string) - Backend used by InSpec for connetction. Defaults to
+    SSH.
+
+-   `host` (string) - Host used for by InSpec for connection. Defaults to
+    localhost.
```

Comment:
not sure how WinRM works but for the InSpec docker-target this would be the container id.

Reply:
well than i think this is not needed, except WinRM is somehow special

Reply:
Another use case might be a static IP-address. But other than that I'm not sure. Thought maybe somebody else could find it useful to have as a configurable option.

Reply:
The container name (specified via `--name`) also works.

Reply:
What happens with docker if no name/host is specified?

Reply:
It would try to use localhost.
```plaintext
2019-02-12T08:17:44Z: ==> docker: Provisioning with Inspec...
2019-02-12T08:17:44Z: ==> docker: Executing Inspec: inspec exec /home/vagrant/test.rb --backend docker --host 127.0.0.1 --attrs /tmp/packer-provisioner-inspec.627871494.yml
2019-02-12T08:17:47Z:     docker: /var/lib/gems/2.5.0/gems/docker-api-1.34.2/lib/docker/connection.rb:46:in `rescue in request': No such container: 127.0.0.1 (Docker::Error::NotFoundError)
```

Reply:
Do you think it is better to add a check that `host` must be specified when the `backend` is docker?

Reply:
Definitely have that check if it'll break when unset. 👍

Reply:
Done